harness/determined/metrics.py

```diff
@@ -0,0 +1,454 @@
+                self.current_metrics.reset()
+                batch_start_time = time.time()
+
+            time.sleep(0.02)
+
+    def update_current_batch(self, new_current_batch):
+        self.current_batch = new_current_batch
+
+    def enqueue_for_async_send(self, metric_batch):
+        # This method can theoretically raise a FULL error, but SimpleQueues are unbounded so
+        # I don't think it will ever happen (https://docs.python.org/3/library/queue.html#queue.Queue.put)
+        # self.log("Enqueuing metric batch", metric_batch)
+        self.dispatch_queue.put_nowait(metric_batch)
+
+    def __enter__(self) -> "SystemMetricsThread":
+        self.start()
+        return self
+
+    def __exit__(self, *arg: Any) -> None:
+        self.quitting = True
+
+
+
+
+
+
+
+# This is a thread that exists solely so that we can make API calls without blocking
+# It has a SimpleQueue through which work is sent to the thread
+class SystemMetricsSendingThread(threading.Thread):
+    def __init__(self, inbound_queue: queue.Queue, master_url: str) -> None:
+        print("[SystemMetricsSendingThread] Creating SystemMetricsSendingThread")
+        self.master_url = master_url
+        self.POLL_INTERVAL_SECS = 0.5
+        self.inbound_queue = inbound_queue
+
+        self.short_circuit_counter = 0
+        self.all = []
+
+        self.quitting = False
+        super().__init__()
+
+    def run(self) -> None:
+        while True:
+            if self.quitting:
+                break
+
+            try:
+                batch_to_send = self.inbound_queue.get_nowait()
+            except queue.Empty:
+                time.sleep(self.POLL_INTERVAL_SECS)
+                continue
+
+            # print(f"[SystemMetricsSendingThread] Sending a batch. {humanize_float(time.time())}")
+            self.send_batch(batch_to_send)
+
+    # This is a blocking operation that must handle all exceptions gracefully
+    def send_batch(self, post_bodies):
+        # TODO: Handle exceptions and automatically retry X times and then log and drop the batch
+        # TODO: Change API so we don't have to make so many independent API calls
+
+        for post_body in post_bodies:
+            api.post_trial_profiler_metrics(
+                self.master_url,
+                post_body["values"],
+                post_body["batches"],
+                post_body["timestamps"],
+                post_body["labels"],
+            )
+
+    def quit(self):
+        self.quitting = True
+
+
+
+
+
+GIGA = 1_000_000_000
+
+class SimpleCpuUtilCollector:
+    def measure(self, batch_idx):
+        timer = QuickTimer("SimpleCpuUtilCollector")
+        cpu_util = psutil.cpu_percent()
+        timestamp = datetime.datetime.utcnow()
+        timer.stop()
+        return Measurement(timestamp, batch_idx, cpu_util)
+
+
+
+class FreeMemoryCollector:
+    # We choose to report free memory instead of available memory because it is useful to
```

Comment:
I would like us not to show memory usage for cached files because I think that's almost always unhelpful.  We've had customers whose machines would look like they were teetering on OOM when in reality the kernel was just caching a lot of files, and cached files never cause OOM (see https://www.linuxatemyram.com/).

Personally I found psutil.memory_percent() to be pretty useful; not quite sure how that's derived from system values though.

Reply:
Okay, I'll switch it to available. Cached file usage can be helpful for analyzing performance, but only if you can break it out as a single number.